### PR TITLE
Assisted Installer deployment module

### PR DESCRIPTION
**Pull request by dahorak**
This PR depends on https://github.com/red-hat-storage/ocs-ci/pull/8994/

### CODE REVIEW

**Comment by dahorak** [comment]
It is not possible to test this module via Jenkins job, but I've tested it together with changes from [PR#8994](https://github.com/red-hat-storage/ocs-ci/pull/8994/files) and other changes related to the deployment via Assisted Installer on vSphere locally and it worked as expected, so I would consider it as verified.

File: ocs_ci/deployment/assisted_installer.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+        """
+        Wait for expected number of nodes to appear in the Assisted Installer infra/cluster
+
+        Args:
+            expected_nodes (int): number of expected nodes
+        """
+
+        # wait for discovered nodes in cluster definition
+        for sample in TimeoutSampler(
+            timeout=3600, sleep=300, func=self.api.get_cluster_hosts, cluster_id=self.id
```
Comment (DanielOsypenko) [inline ocs_ci/deployment/assisted_installer.py:227]: I think it worth to carry the timeour to arguments

File: ocs_ci/deployment/assisted_installer.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+        # create metadata, kubeconfig, kubeadmin-password and openshift_install.log files
+        self.create_metadata_file()
+        self.create_kubeconfig_file()
+        self.create_kubeadmin_password_file()
+        self.create_openshift_install_log_file()
+
+    def create_metadata_file(self):
+        """
+        Create .openshift_install.log file containing URL to OpenShift console
+        It is used by our CI jobs to show the console URL in build description.
```
Comment (DanielOsypenko) [inline ocs_ci/deployment/assisted_installer.py:331]: probably description here miscopied from the next func create_openshift_install_log_file

**Review by DanielOsypenko — DISMISSED** [review]

**Review by DanielOsypenko — DISMISSED** [review]

**Review by petr-balogh — APPROVED** [review]

**Comment by dahorak** [comment]
Just re-based on top of master and resolved conflict.

**Review by petr-balogh — APPROVED** [review]

**Review by DanielOsypenko — APPROVED** [review]

**Pull request merged**